### PR TITLE
Rd 2515: Added support to add realm to member

### PR DESCRIPTION
**Pull request by Prateek-Gupta1**

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -3,6 +3,6 @@ rootProject.name='sdk'
 include 'core'
 include 'user'
 include 'tpp'
-include 'android'
+//include 'android'
```
Comment (yuhanniu) [inline settings.gradle]: revert

File: tpp/src/main/java/io/token/tpp/TokenClient.java [hunk 2]
```diff
@@ -115,18 +115,32 @@ public static TokenClient create(TokenCluster cluster, String developerKey) {
...
 
+    /**
+     * Creates a new Token member with a set of auto-generated keys, an alias, and member type.
+     *
+     * @param alias nullable member alias to use, must be unique. If null, then no alias will
+     *     be created with the member.
+     * @param partnerId ID of partner member.
+     * @return newly created member
+     */
+    public Observable<Member> createMember(final Alias alias, @Nullable String partnerId) {
```
Comment (yuhanniu) [inline tpp/src/main/java/io/token/tpp/TokenClient.java]: this `@Nullable` is redundant, if it is null then `createMember(final Alias alias)` should be called

File: tpp/src/main/java/io/token/tpp/TokenClient.java [hunk 3]
```diff
@@ -167,6 +181,33 @@ public Member createMemberBlocking(Alias alias, String partnerId) {
...
+    /**
+     * Creates a new Token member in the provided realm with a set of auto-generated keys, an alias,
+     * and member type.
+     *
+     * @param alias nullable member alias to use, must be unique. If null, then no alias will
+     *     be created with the member.
+     * @param realmId member id of an existing Member to whose realm this new member belongs.
+     * @return newly created member
+     */
+    public Observable<Member> createMemberInRealm(final Alias alias, @Nullable String realmId) {
```
Comment (yuhanniu) [inline tpp/src/main/java/io/token/tpp/TokenClient.java]: just `createMember` to be consistent, this @Nullable is redundant.

File: user/src/main/java/io/token/user/TokenClient.java [hunk 4]
```diff
@@ -129,7 +128,19 @@ public static TokenClient create(TokenCluster cluster, String developerKey) {
...
+
+    /**
+     * Creates a new Token member with a set of auto-generated keys, an alias, and member type.
+     *
+     * @param alias nullable member alias to use, must be unique. If null, then no alias will
+     *     be created with the member.
+     * @param recoveryAgent member id of the primary recovery agent.
+     * @return newly created member
+     */
+    public Observable<Member> createMember(final Alias alias, @Nullable String recoveryAgent) {
```
Comment (yuhanniu) [inline user/src/main/java/io/token/user/TokenClient.java]: @Nullable redundant

File: user/src/main/java/io/token/user/TokenClient.java [hunk 5]
```diff
@@ -177,13 +190,40 @@ public Member createMemberBlocking(Alias alias) {
...
+    /**
+     * Creates a new Token member in the provided realm with a set of auto-generated keys, an alias,
+     * and member type.
+     *
+     * @param alias nullable member alias to use, must be unique. If null, then no alias will
+     *     be created with the member.
+     * @param realmId member id of the Member whose realm this new Member belongs.
+     * @return newly created member
+     */
+    public Observable<Member> createMemberInRealm(final Alias alias, @Nullable String realmId) {
```
Comment (yuhanniu) [inline user/src/main/java/io/token/user/TokenClient.java]: `createMember`, remove Nullable

File: tpp/src/main/java/io/token/tpp/TokenClient.java (hunk 3 above)
Comment (Prateek-Gupta1) [reply tpp/src/main/java/io/token/tpp/TokenClient.java]: `createMember(string, string)` already exists for adding partnerId.

File: tpp/src/main/java/io/token/tpp/TokenClient.java [hunk 6]
```diff
@@ -167,6 +182,33 @@ public Member createMemberBlocking(Alias alias, String partnerId) {
...
+    /**
+     * Creates a new Token member in the provided realm with a set of auto-generated keys, an alias,
+     * and member type.
+     *
+     * @param alias nullable member alias to use, must be unique. If null, then no alias will
+     *     be created with the member.
+     * @param realmId member id of an existing Member to whose realm this new member belongs.
+     * @return newly created member
+     */
+    public Observable<Member> createMemberInRealm(final Alias alias, String realmId) {
```
Comment (yuhanniu) [inline tpp/src/main/java/io/token/tpp/TokenClient.java]: we can have a `createMemberInRealm` in the user sdk too, and behavior should be we also use the realmId as the recovery agent.

File: user/src/main/java/io/token/user/TokenClient.java [hunk 7]
```diff
@@ -140,7 +140,7 @@ public static TokenClient create(TokenCluster cluster, String developerKey) {
      * @return newly created member
      */
     public Observable<Member> createMember(final Alias alias, String recoveryAgent) {
-        return createMember(alias, recoveryAgent, null);
+        return createMember(alias, recoveryAgent, recoveryAgent);
```
Comment (yuhanniu) [inline user/src/main/java/io/token/user/TokenClient.java]: recovery agent is not necessarily the realm

**Pull request merged**